src/main/java/com/github/hokkaydo/eplbot/module/code/RustCompiler.java

```diff
@@ -0,0 +1,129 @@
+
+    private static final ScheduledExecutorService SCHEDULER = new ScheduledThreadPoolExecutor(1);
+    @Override
+    public String run(String input, Integer runTimeout) {
+        if (containsUnsafeKeywords(input)){
+            return "Compilation failed:\nCheck if 'std' or 'use' are used";
+        }
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        File sourceFile;
+        Process compile;
+        try {
+
+            sourceFile = new File(CURRENT_DIR, "temp.rs");
+            sourceFile.deleteOnExit();
+            FileWriter writer = new FileWriter(sourceFile);
+            writer.write(input);
+            writer.close();
+            compile = new ProcessBuilder("rustc", sourceFile.getAbsolutePath()).directory(new File(CURRENT_DIR)).redirectErrorStream(true).start();
+
+        } catch (IOException e) {
+            e.printStackTrace();
+            return "Server error code R01" + e.toString();
+        }
+        int compileExitCode;
+        try{
+            compileExitCode = compile.waitFor();
+        }catch (InterruptedException e){
+            e.printStackTrace();
+            return "Server error code R02" + e.toString();
+        }
+        BufferedReader reader = new BufferedReader(new InputStreamReader(compile.getInputStream()));
+        String line;
+        try {
+            while ((line = reader.readLine()) != null) {
+                outputStream.write((line + "\n").getBytes());
+            }
+        } catch (IOException e){
+            e.printStackTrace();
+            return "Server error code R03" + e.toString();
+        }
+
+        if (compileExitCode != 0) {
+            deleteFiles();
+            return "Compilation failed:\n" + outputStream;
+        }
+        String executableFileName = sourceFile.getName().replace(".rs", "");
+        File executableFile = new File(CURRENT_DIR, executableFileName);
+
+        if (!executableFile.setExecutable(true)) {
+            return "Couldn't make file executable";
+        }
+        Process run;
+        try {
+            run = new ProcessBuilder(new File(CURRENT_DIR, executableFileName).getAbsolutePath()).directory(new File(CURRENT_DIR)).redirectErrorStream(true).start();
+        } catch (IOException e){
+            e.printStackTrace();
+            return "Server error code R04" + e.toString();
+        }
+
+        BufferedReader reader2 = new BufferedReader(new InputStreamReader(run.getInputStream()));
+        ScheduledFuture<?> timeOut = SCHEDULER.schedule(() -> {}, runTimeout, TimeUnit.SECONDS);
+        String line2;
+        try {
+            while ((line2 = reader2.readLine()) != null) {
+                outputStream.write((line2 + "\n").getBytes());
+            }
+        } catch (IOException e){
+            e.printStackTrace();
+            Thread.currentThread().interrupt();
+            return "Server error code R05" + e.toString();
```

Comment:
No need to give the stack trace to the user, the error code should be enough for us, devs, to identify the stack trace in the logs
```suggestion
            return "Server error code R05" + e;
```